0000-navigator-view-api.md

```diff
@@ -0,0 +1,93 @@
+* Start Date: 2018-01-29
+* RFC PR: (leave this empty)
+* React Navigation Issue: (leave this empty)
+
+# Summary
+
+Introduce a simpler API for createNavigator and navigation views. By providing "sceneDescriptors" to the navigation view, it does not need to compute options or manually create the child navigation prop.
+
+Down the line, this will enable us to deprecate `addNavigationHelpers`, which is an avoidable burden for redux apps and when implementing custom navigators. Eventually, each navigator may provide a different set of helpers.
+
+# Basic example
+
+Using a new `createNavigator` API and `screenDescriptors` prop, this is simplest possible custom navigation view:
+
+```js
+const CustomNavigationView = ({navigation, screenDescriptors}) => {
+  const {routes, index}
+  const descriptor = screenDescriptors.find(d => d.key === routes[index].key);
+  const ScreenComponent = descriptor.getComponent();
+  return (
+    <ScreenComponent
+      navigation={descriptor.navigation}
+      options={descriptor.options}
+    />
+  );
+};
+const router = StackRouter({...}, {...});
+const CustomNavigator = createNavigator(CustomNavigationView, router);
+```
+
+This `<CustomNavigator />` will have behave like a stack, and will be able to present screens registered within `router`.
```

Comment:
probably worth being explicit about what this navigation view would do (it would only render the currently active screen)

Reply:
sounds good